tensorflow_io/numpy/python/ops/numpy_ops.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Numpy Dataset."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+import tensorflow as tf
+from tensorflow_io.core.python.ops import core_ops
+from tensorflow_io.core.python.ops import data_ops
+
+class _NumpyDataset(data_ops.BaseDataset):
+  """NumpyDataset"""
+  def __init__(self, array, filename=None, **kwargs):
+    if filename is not None:
+      arrayname = array.name if array.name is not None else ""
+      address = 0
+      arrayshape = array.shape.as_list()
+      dtype = array.dtype
+      shape = tf.TensorShape(
+          [dim if i != 0 else None for i, dim in enumerate(
+              array.shape.as_list())])
+      start = 0
+      stop = array.shape[0]
+    else:
+      self._array_data = array.data
```

Comment:
I believe `array.data` is a memoryview, does that increment the refcount of the object?